ale_linters/cuda/nvcc.vim

```diff
@@ -0,0 +1,61 @@
+" Author: blahgeek <i@blahgeek.com>
+" Description: NVCC linter for cuda files
+
+call ale#Set('cuda_nvcc_executable', 'nvcc')
+call ale#Set('cuda_nvcc_options', '-std=c++11')
+
+function! ale_linters#cuda#nvcc#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'cuda_nvcc_executable')
+endfunction
+
+function! ale_linters#cuda#nvcc#GetCommand(buffer) abort
+    let l:output_file = tempname() . ".ii"
+    call ale#engine#ManageFile(a:buffer, l:output_file)
+    return ale#Escape(ale_linters#cuda#nvcc#GetExecutable(a:buffer))
+    \   . ' -cuda '
+    \   . ale#c#IncludeOptions(ale#c#FindLocalHeaderPaths(a:buffer))
+    \   . ale#Var(a:buffer, 'cuda_nvcc_options') . ' %s'
+    \   . ' -o ' . l:output_file
```

Comment:
I wonder, is there an option to cancel output, or output to a null file? Most tools like this have such an option.

Reply:
If "-o xxx" is omitted, it would write output to current directory. It can output to the null (/dev/null), but it cannot work on windows, am i right?

Reply:
Does `-o nul` work on Windows? There's a variable in `ale#util` for giving you the path to the null file for each platform.

Reply:
I'd explore that option first, so no files can be generated at all. If that doesn't work out, then ALE has some tools for managing temporary files so they can be automatically deleted when linting is done, which were built for handling this kind of problem.

Reply:
I've changed the code to use `ale#util#nul_file` and it works on linux, but I am unable to test on windows for now.